src/renderer/window-setup.js

```diff
@@ -99,23 +99,46 @@ module.exports = (win, ipcRenderer, guestInstanceId, openerId, nonNativeWinOpenF
     };
   }
 
-  if (openerId != null && win.opener == null) {
-    win.opener = getOrCreateProxy(ipcRenderer, openerId);
-  }
+  if (win.opener === null) {
+    if (openerId !== null) {
+      win.opener = getOrCreateProxy(ipcRenderer, openerId);
+    }
 
-  ipcRenderer.removeAllListeners('ELECTRON_GUEST_WINDOW_POSTMESSAGE');
-  ipcRenderer.on('ELECTRON_GUEST_WINDOW_POSTMESSAGE', (event, sourceId, message, sourceOrigin) => {
-    event = new MessageEvent('message', { data: message, origin: sourceOrigin });
-    event.source = getOrCreateProxy(ipcRenderer, sourceId);
+    // Optimistic code block workaround for a well-know Electron bug
+    //
+    // After cross site navigation with the native window.open method,
+    // we lost the opener reference.
+    //
+    // We polyfill the *postMessage* method to forward the message via the main
+    // (src/browser/engine/window.ts). The main broadcast the message for
+    // all allowed webContents (matching type, protocol) only if the
+    // webContents sender match the *targetOrigin*
+    //
+    // used by: Mixmax and other apps using Google OAuth
+    // issue ref: https://github.com/electron/electron/issues/18032
+    win.opener = {
```

Comment:
After the fix of electron/electron#18032 in electron 5.0.4, I think we can remove most of these workarounds, no?

Reply:
yep I on it

Reply:
🆙